app/controllers/users_controller.rb

```diff
@@ -21,17 +21,23 @@ def show
     end
   end
 
+  # rubocop: disable Metrics/MethodLength
   def create
-    @user = User.new(user_params)
-    @user.provider = 'local'
-    if @user.save
-      @user.send_activation_email
-      flash[:info] = 'Please check your email to activate your account.'
-      redirect_to root_url
+    @user = User.find_by email: user_params[:email].downcase
```

Comment:
You could use a case-insensitive index without PostgreSQL-specific stuff - and you should probably investigate that alternative just to see if it's better.  But you'd need to make sure the case-insensitive index is actually *used*.